appservice/src/FileUtilities.ts

```diff
@@ -40,30 +27,67 @@ export async function zipFile(filePath: string): Promise<string> {
     return zipFilePath;
 }
 
-export async function zipDirectory(folderPath: string, globPattern: string = '**/*', ignorePattern?: string | string[]): Promise<string> {
+async function zipDirectoryInternal(folderPath: string, addFiles: (zipper: archiver.Archiver) => Promise<void>): Promise<string> {
     if (!folderPath.endsWith(path.sep)) {
         folderPath += path.sep;
     }
 
     const zipFilePath: string = path.join(os.tmpdir(), `${randomFileName()}.zip`);
-    await new Promise((resolve: () => void, reject: (err: Error) => void): void => {
-        const zipOutput: fs.WriteStream = fs.createWriteStream(zipFilePath);
+    await new Promise(async (resolve: () => void, reject: (err: Error) => void): Promise<void> => {
+        const zipOutput: fse.WriteStream = fse.createWriteStream(zipFilePath);
         zipOutput.on('close', resolve);
 
         const zipper: archiver.Archiver = archiver('zip', { zlib: { level: 9 } });
         zipper.on('error', reject);
+        await addFiles(zipper);
         zipper.pipe(zipOutput);
-        zipper.glob(globPattern, {
-            cwd: folderPath,
-            dot: true,
-            ignore: ignorePattern
-        });
         void zipper.finalize();
     });
 
     return zipFilePath;
 }
 
+/**
+ * Zips directory using glob filtering
+ */
+export async function zipDirectoryGlob(folderPath: string, globPattern: string = '**/*', ignorePattern?: string | string[]): Promise<string> {
+    return await zipDirectoryInternal(folderPath, async (z) => {
+        await addFilesByGlob(z, folderPath, globPattern, ignorePattern);
+    });
+}
+
+/**
+ * Zips directory using gitignore filtering
+ */
+export async function zipDirectoryGitignore(folderPath: string, gitignoreName: string): Promise<string> {
+    return await zipDirectoryInternal(folderPath, async (z) => {
+        await addFilesByGitignore(z, folderPath, gitignoreName);
+    });
+}
+
+async function addFilesByGlob(zipper: archiver.Archiver, folderPath: string, globPattern: string, ignorePattern: string | string[] | undefined): Promise<void> {
+    zipper.glob(globPattern, {
+        cwd: folderPath,
+        dot: true,
+        ignore: ignorePattern
+    });
+}
+
+async function addFilesByGitignore(zipper: archiver.Archiver, folderPath: string, gitignoreName: string): Promise<void> {
```

Comment:
Should we ignore files in the `.gitignore`? I think we discussed this before, but couldn't some of these files be needed for deployment?

Though I see you're using it for `.funcignore` but that seems different to me.

Reply:
It's not actually using the gitignore - it's just using the _filtering rules_ of a gitignore. gitignore has it's own patterns that are unique and not exactly the same as glob: https://git-scm.com/docs/gitignore#_pattern_format

Reply:
No, I know, I was questioning the naming convention of your function 😋

Reply:
You're welcome to suggest a name that makes more sense to you :)